### DISCUSSION TITLE
Writing to a S3 Bucket with Server-side encryption enabled

### DISCUSSION THREAD

**Discussion by alborz-plerionaut** [delta-io/delta-rs#1612] [Q&A]
To upload objects to an S3 bucket that utilizes AWS Key Management Service (SSE-KMS) for server-side encryption, we need to include the HTTP header {"s3:x-amz-server-side-encryption": "aws:kms"} in the request. Is there a method available to achieve this?

**Comment by rtyler** [comment]
I was looking briefly at the [object_store](https://docs.rs/object_store/0.7.0/object_store/aws/struct.AmazonS3Builder.html) documentation, and I'm not sure there's a good way to specify the server side encryption. I _thought_ that S3 buckets with encryption set by default wouldn't require this.

Are you wanting to specify a custom KMS key?  If we can find a way to get what you need from `object_store` then it's a relatively straightforward exercise of constructing the `ObjectStore` instance and passing it into the `deltalake` code

**Comment by mightyshazam** [comment]
One way to do it is to use a lambda that updates the metadata of the file after writing it.
